lib/js-file.js

```diff
@@ -244,26 +185,60 @@ JsFile.prototype = {
     getLastNodeToken: function(node) {
         return this.getTokenByRangeEnd(node.range[1]);
     },
+
     /**
      * Returns the first token before the given.
      *
      * @param {Object} token
+     * @param {Object} [options]
+     * @param {Boolean} [options.includeComments=false]
      * @returns {Object|undefined}
      */
-    getPrevToken: function(token) {
+    getPrevToken: function(token, options) {
         var index = token._tokenIndex - 1;
-        return index >= 0 ? this._tree.tokens[index] : undefined;
+        if (index < 0) {
+            return undefined;
+        }
+
+        if (options && options.includeComments) {
+            return this._tokens[index];
+        }
+
+        do {
+            if (!this._tokens[index].isComment) {
+                return this._tokens[index];
+            }
+        } while (--index >= 0);
+
+        return undefined;
     },
+
     /**
      * Returns the first token after the given.
      *
      * @param {Object} token
+     * @param {Object} [options]
+     * @param {Boolean} [options.includeComments=false]
      * @returns {Object|undefined}
      */
-    getNextToken: function(token) {
+    getNextToken: function(token, options) {
         var index = token._tokenIndex + 1;
-        return index < this._tree.tokens.length ? this._tree.tokens[index] : undefined;
+
+        if (index >= this._tokens.length) {
+            return undefined;
+        }
+
+        if (options && options.includeComments) {
+            return this._tokens[index];
+        }
+
+        do {
+            if (!this._tokens[index].isComment) {
+                return this._tokens[index];
+            }
+        } while (++index < this._tokens.length);
```

Comment:
@mikesherov At least we can run tests before and after that PR.
I know there are a couple of additional tests but...
Executing `time mocha` in `master`:
```
real    0m13.345s
user    0m12.157s
sys     0m0.304s
```
And in `mdevils.autofixing-token-data`:
```
real    0m21.480s
user    0m19.745s
sys     0m0.528s
```
It's like 1.5x slower ;-(